app/Services/UpdateUserService.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\Facades\Auth;
+
+class UpdateUserService
+{
+    public function updateUser(array $userData)
+    {
+        $user = Auth::user();
+
+        if (!empty($userData['new_password'])) {
+            $user->password = $userData['new_password'];
+        }
+        if (!empty($userData['email'])) {
+            $user->email = $userData['email'];
+        }
+
+        if ($user->can('update', $user)) {
```

Comment:
move this check to your request authorize method

return $user->can('update', $this->user)